tracker/progress.go

```diff
@@ -35,8 +35,16 @@ type Progress struct {
 	// entries with indices in (Match, Next) interval are already in flight.
 	//
 	// Invariant: 0 <= Match < Next.
+	// NB: it follows that Next >= 1.
+	//
+	// In StateSnapshot, Next == PendingSnapshot + 1.
 	Next uint64
 
+	// sentCommit is the highest commit index in flight to the follower.
```

Comment:
```suggestion
	// sentCommit is the highest commit index in flight to the follower.
	// It may be regressed to `Next -1` in some cases, e.g. when converting
	// to `StateProbe` or when receiving a rejection from a follower. 
```

Reply:
Reworded this a bit.